### ISSUE TITLE
gnus faces changed by modus-vivendi theme

### ISSUE THREAD

**Issue by sivaramn** [protesilaos/modus-themes#119]
In emacs 29.4, modus-themes 4.6+ version, I use modus-vivendi as my theme.  It all works very well, except in Gnus, where Summary marks are not respected(read, unread, marked as spam etc) in terms of their faces. Is there a way to customize the faces of gnus messages of Group and Summary buffers?  In the screenshot, all the messages are read but it looks like all are unread entries.  

![image](https://github.com/user-attachments/assets/afbbb67e-c5f6-4aed-8ef1-c7761d0f2b8f)

another example, the entire summary buffer is seen but it looks as if so many are unread. I opened only 2 articles.
![image](https://github.com/user-attachments/assets/dededaaf-37e6-4548-b5dd-5211f94e195a)

Basically, what should I customize to get it to look like a normal inbox folder? this section works for the mail headers when I open the article
https://protesilaos.com/emacs/modus-themes#h:7da7a4ad-5d3a-4f11-9796-5a1abed0f0c4

**Comment by protesilaos** [comment]
> From: sivaramn ***@***.***>
> Date: Sat,  2 Nov 2024 02:01:16 -0700
>
> [... 5 lines elided]
>
> In the screenshot, all the messages are read but it looks like all are
> unread entries.

Can you place the cursor of the text and do 'M-x describe-char'? It will
produce a Help buffer that will mention the name of the face in use.
Tell me which one it is and I will update the themes accordingly.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by sivaramn** [comment]
> From: sivaramn ***@***.***> Date: Sat, 2 Nov 2024 02:01:16 -0700 [... 5 lines elided] In the screenshot, all the messages are read but it looks like all are unread entries.
> Can you place the cursor of the text and do 'M-x describe-char'? It will produce a Help buffer that will mention the name of the face in use. Tell me which one it is and I will update the themes accordingly.
> […](#)
> -- Protesilaos Stavrou https://protesilaos.com

Placing it on any unread mail subj and doing as suggested gives this

_There is an overlay here:
 From 9332 to 9408
  face                 hl-line
  priority             -50
  window               #<window 3 on *Summary nntp+news.gmane.io:gmane.emacs.devel*>

There are text properties here:
  face                 gnus-summary-normal-ancient
  gnus-number          325043
  mouse-face           highlight_

on a related note, the group buffer looks like this

![image](https://github.com/user-attachments/assets/6feb8e0f-92f5-45bc-acf0-c78316f96414)

with describe-char showing

There is an overlay here:
 From 119 to 173
  face                 hl-line
  priority             -50
  window               #<window 3 on *Group*>


There are text properties here:
  face                 gnus-group-news-3
  mouse-face           highlight

**Comment by protesilaos** [comment]
This is a tricky case. The face you found there is `gnus-summary-normal-ancient`, but your problem is with read VS unread. I will need to set up Gnus to try this myself.

How do those buffers look if you do `M-x disable-theme`? Is there something else involved other than `gnus-summary-normal-ancient`?

**Comment by protesilaos** [comment]
Thank you!

Can you please load `modus-vivendi` and then evaluate the following?

```elisp
(modus-themes-with-colors
  (custom-set-faces
   `(gnus-summary-normal-ancient ((,c :foreground ,fg-alt)))))
```

I am curious if it improves things on your end.

**Comment by protesilaos** [comment]
> which is a better; appears less bright but not by much. 

Yes, it is better but I think it is not good enough. I will need to reveiw Gnus, though this is not an easy project.

> with a systemic theme for mail handling from spam,muted, less important, normal, important mail rows in summary buffer

At the theme level we can only cover whatever the package defines as a face. If there are faces for all the cases you describe, then I am happy to consider them. I believe the themes already cover all of the faces defined by Gnus, notwithstanding the specific face we are not testing.

**Comment by protesilaos** [comment]
Hello again @sivaramn!

I am trying to set up to Gnus to better understand the interplay of certain faces. I am browsing the gmane groups and trying to load old ("ancient") posts, such as from the Emacs development group. But even if I load the last 100000 messages, I still do not get anything that uses the `gnus-summary-ancient-normal`. Anyway, I was using this to easily spot the messages:

```elisp
(modus-themes-with-colors
  (custom-set-faces
   `(gnus-summary-normal-ancient ((,c :foreground ,red)))))
```

**Comment by sivaramn** [comment]
this looks a bit angry TBH

![Image](https://github.com/user-attachments/assets/4082d2e2-af10-4c75-97af-247bb85361cb)


In the gnus help documentation there's a section `3.1.3 Group Highlighting`  `4.18.1 Article Highlighting`  `10.7 Faces and Fonts` which might be interfering with what you've got?

**Comment by protesilaos** [comment]
> this looks a bit angry TBH

Oh no, sorry! This is not my intended style. It is just easier for me to spot it and then do the experiment.

I think I know what is happening. On your screen you have the `O` mark next to those threads, whereas on mine it is not there. I guess this had to do with the messages being read. I will check around.

More to follow.

**Comment by protesilaos** [comment]
I finally managed to reproduce the issue. I decided to make those `O` marked articles the same as the `R` or `M` ones. They all seem to denote "already read this", but the Gnus manual and the source code are not clear about the semantics.

Let me know if you think this is an appropriate design.

**Comment by sivaramn** [comment]
![Image](https://github.com/user-attachments/assets/9a37ce97-8248-4f9f-ae65-5684e1ce1a51)

with themes disabled but the red colour kept, it looks like this.  And gnus does change the face if the article is marked as spam, low/high priority or frequently read poster as per the styles i listed earlier.

What should I test now?

**Comment by protesilaos** [comment]
> with themes disabled but the red colour kept, it looks like this.

The red colour should be removed. The code I sent you earlier which was making them red was just for demo purposes: red was an easy colour for me to quickly spot the effect. But it was never the intended style: those "ancient" messages should be subtle.

> And gnus does change the face if the article is marked as spam, low/high priority or frequently read poster as per the styles i listed earlier.

The themes already cover the relevant faces, though I am interested to know if there is some scenario where you think the style does not look right. I am happy to make changes.

**Comment by protesilaos** [comment]
Just to be clear, the current style of the "ancient" messages should be a subtle grey foreground. Same as that of read messages (those which have the `R` mark next to them).

**Comment by protesilaos** [comment]
You are welcome!
